### DISCUSSION TITLE
Cannot install prisma engines

### DISCUSSION THREAD

**Discussion by Trapsen** [prisma/prisma#24595] [Q&A] [answered]
### Bug description

Hello, 
Unfortunately, I cannot install Prisma Engines on my Linux system. The following error appears:
` Downloading Prisma engines for Node-API for debian-openssl-1.1.x [                    ] 0%Error: request to https://binaries.prisma.sh/all_commits/5675a3182f972f1a8f31d16eee6abf4fd54910e3/debian-openssl-1.1.x/libquery_engine.so.node.gz.sha256 failed, reason: self-signed certificate in certificate chain`

This is a company server and I cannot change the network. On my PC I can download the engines. Is it possible, to install them manually?

### Prisma information

```prisma
// Add your schema.prisma
```

```ts
// Add your code using Prisma Client
```

### Environment & setup

- OS: Ubuntu
- Database: SQLServer
- Node.js version: v20.14.0

### Prisma Version

```
not available
```

**Comment by ludralph** [comment]
Hi @Trapsen 👋 

Thank you for raising this question.

One alternative to downloading the query engine is to build it on your machine. You can build the Prisma Engine for your Linux variant by following the instructions in this [Readme](https://github.com/prisma/prisma-engines/blob/main/README.md#building-prisma-engines) file.

Another alternative is that you can also transfer the downloaded query engine from your PC to the company server by copying it via USB or SCP.

**Reply by Trapsen** [reply]
Hi @ludralph ,

Thank you for your answer.

If I download them on my pc, where do I need to move them in the project, that the engines will be recognized?

**Reply by ludralph** [reply]
Hi @Trapsen 

You can then set environment variables to point Prisma to the location of these binaries on your server. See the using custom engine libraries section in the [documentation](https://www.prisma.io/docs/orm/more/under-the-hood/engines#using-custom-engine-libraries-or-binaries).

**Reply by Trapsen** [reply]
Hi @ludralph 

Thank you. I also recognized, that prisma tries to download the .sha256 files. Do I also need to point to them in the enviroment variables? Or can I skip them?

**Reply by ludralph** [reply, marked as answer]
Hi @Trapsen 

You can skip them. The .`sha256` checksum files are used to verify the integrity of the engine binaries. To skip the checksum validations, you can use the `PRISMA_ENGINES_CHECKSUM_IGNORE_MISSING` environment variable. Setting this variable to a truthy value will instruct Prisma to ignore problems related to downloading and verifying the integrity of the Prisma engines via checksum files. See this [section](https://www.prisma.io/docs/orm/reference/environment-variables-reference#prisma_engines_checksum_ignore_missing) of the documentation.

**Reply by Trapsen** [reply]
Hi @ludralph 

Thank you. In my .env file I have set the `PRISMA_ENGINES_CHECKSUM_IGNORE_MISSING=1` and it doesn't work. The same error as in the beginning appears.

**Reply by Trapsen** [reply]
Hi @ludralph 

I was able to find the error. I have now set the enviroment variables via export [variable name] and now it works.

Thank you!
